packages/hardhat/contracts/NFTAuctionMachine.sol

```diff
@@ -210,3 +226,42 @@ contract NFTAuctionMachine is ERC721, Ownable, ReentrancyGuard, JBETHERC20Projec
             super.supportsInterface(interfaceId);
     }
 }
+
+// Metadata contract where all NFTs get same metadata
+contract SingleUriMetadata is IMetadata, Ownable {
+    event URIChanged(string indexed newURI);
+
+    string public uri;
+    bool metadataFrozen;
```

Comment:
as mentioned above we can make this public and read this right to avoid consuming more storage space in auction contract ?

Reply:
public yes, but hmmmm.. there's a difference between freezing the metadata uri and freezing the ability to change the metadata contract altogether isnt there?